src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,107 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private Object[] data = new Object[DEFAULT_CAPACITY];
+
     @Override
     public void add(T value) {
-
+        if (size == data.length) {
+            data = grow();
+        }
+        data[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        indexRangeCheckForAdd(index);
+        if (size == data.length) {
+            data = grow();
+        }
+        System.arraycopy(data, index, data, index + 1, size - index);
+        data[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
```

Comment:
You can use for each loop here

Reply:
I faced "Foreach not applicable to type 'core.basesyntax.List<T>" message while trying to make for each loop. I thought that's because we don't implement iterator method (nothing was said about that in the task description).  Should I implement this method or am I generally wrong?